app/controllers/fields_controller.rb

```diff
@@ -0,0 +1,30 @@
+class FieldsController < AuthenticatedController
+  # Returns the form view given a source text
+  def form
+    @form_data = HasFields.parse_fields(params[:source])
+    @allow_dropdown = params[:allow_dropdown] == 'true'
+  end
+
+  # Returns a single field for the form view
+  def field
+    @index = params[:index].to_i
+  end
+
+  # Returns the source text given a form data
+  def source
+    render plain: convert_to_source
+  end
+
+  private
+
+  # Convert serialized form data to Dradis-style item content
+  def convert_to_source
```

Comment:
As per 

> ... `HasFields#parse_fields`” and it seems this functionality is the opposite transformation (e.g. “give me source from fields”). Those two operations should probably be together.

Seems like the convert to source method should exist in has fields as a class method as well.

Reply:
Maybe as `HasFields.parse_source`

Actually now it feels like the naming is backwards. The two method names should be opposite of each other.
`HasFields.parse_fields` isn't actually parsing fields it's parsing source into fields. So maybe the convert naming is better.

`HasFields.content_to_source` & `HasFields.content_to_fields` or
`HasFields.fields_to_source` & `HasFields.source_to_fields`.